### PR TITLE
fix(sdk): `MutJson` methods `.set()`, `.setAt()` accept `any`, not `MutJson`

### PR DESCRIPTION
**Pull request by revitalbarletz**
Changed `MutJson` methods `.set()`, `.setAt()` to accept `MutJson`  and not
Fixes issue #2767  

## Checklist

- [ ] Title matches [Winglang's style guide](https://docs.winglang.io/contributors/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Comment by revitalbarletz** [comment]
Working on the tests now.

File: examples/tests/sdk_tests/std/json.w [hunk 1]
```diff
@@ -1,2 +1,37 @@
 //-----------------------------------------------------------------------------
-// TODO: https://github.com/winglang/wing/issues/2785
\ No newline at end of file
+// TODO: https://github.com/winglang/wing/issues/2785
+//-----------------------------------------------------------------------------
+bring cloud;
+
+// set & get()
```
Comment (eladb) [inline examples/tests/sdk_tests/std/json.w:6]:
```suggestion
// set() & get()
```

File: examples/tests/sdk_tests/std/json.w [hunk 2]
```diff
@@ -1,2 +1,37 @@
...
+let a = MutJson { a: 1 };
+let b = MutJson { b: 2 };
+a.set("c", b);
+let c = a.get("c");
+assert(c.get("b") == 2);
+
+test "set()" {
+    let x = MutJson { a: 1 };
```
Comment (eladb) [inline examples/tests/sdk_tests/std/json.w:14]: Use two spaces for indentation please

File: examples/tests/sdk_tests/std/json.w [hunk 3]
```diff
@@ -1,2 +1,37 @@
...
+    x.set("b", 2);
+    let y = x.get("b");
+    assert(y == 2);
+}
+
+//-----------------------------------------------------------------------------
+// setAt & getAt()
```
Comment (eladb) [inline examples/tests/sdk_tests/std/json.w:21]:
```suggestion
// setAt() & getAt()
```

**Review by eladb — APPROVED** [review]

**Pull request merged**